src/strategy.py

```diff
@@ -31,9 +30,10 @@ def __init__(self, constants, start_time, **kwargs):
         self.world_state.future_events = []
         self.interval = kwargs.get('interval')  # required
         self.current_time = start_time - self.interval
+        self.count_negative_soc = {}
```

Comment:
count_negative_soc not clear (comment missing). Seems more to keep track of times when SoC is negative than counting? -> rename to negative_soc_tracker?